guides/quickstart.tt

```diff
@@ -0,0 +1,187 @@
+[% WRAPPER layout.tt hideTitle=1 title="Quickstart guide" menu='nixos' %]
+
+<h1>Quickstart guide</h1>
+
+<p class="lead">This quickstart guide will get you started with the Nix package manager.</p>
+
+<p>You will learn how to…</p>
+
+<ul>
+  <li>Install Nix</li>
+  <li>Create your first Nix shell as a working environment</li>
+  <li>Setup a development environment for a web app</li>
+  <li>Build a Docker image to deploy your web app</li>
+</ul>
+
+
+<h2>Install Nix</h2>
+
+<p>While NixOS is a Linux distribution based on Nix, you can install Nix on other Linux distributions, MacOS and Windows via WSL using the install script from our website:</p>
+
+<pre class="code">
+curl -L https://nixos.org/nix/install | sh
+</pre>
+
+<p>(For security, you may want to <a href="[%root%]download.html#nix-verify-installation">verify the script</a> using our GPG signatures.)</p>
+
+<p>Check that the installation was successful by running</p>
+
+<pre class="code">
+nix-channel --list
+</pre>
+
+<p>This command displays the package distribution channel used by Nix. By default, this is <code>https://nixos.org/channels/nixpkgs-unstable</code></p>
+
+
+<h2>Your first shell</h2>
+
+<p>Nix can create reproducible environments given a declarative configuration called a Nix expression. Reproducible means you can share the configuration with others and guarantee that they are using the same software as you.</p>
+
+<p>To get started, make a new folder and create a file called <code>shell.nix</code> with the following contents:</p>
+
+<pre class="code">
+{ pkgs ? import &lt;nixpkgs&gt; {} }:
+
+pkgs.mkShell {
+  buildInputs = [
+    pkgs.which
+    pkgs.htop
+  ];
+}
+</pre>
+
+<p>Basically we import our package channel <code>nixpkgs</code> and make a shell with <code>which</code> and <code>htop</code> as inputs. To enter this environment, type in:</p>
+
+<pre class="code">
+nix-shell
+</pre>
+
+<p>The command will start downloading the missing packages from the cache. This may take a few moments. When it is done, you are dropped into a new shell. This shell provides the packages specified in <code>shell.nix</code>.</p>
+
+<p>Run <code>htop</code> to confirm it is present. Quit the program again by hitting Q.</p>
+
+<p>Now try <code>which htop</code> to check where the <code>htop</code> command is on-disk. You should see something similar to this:</p>
+
+<pre class="code">
+/nix/store/y3w2i8kfdbfj9rx287ad52rahjpgv423-htop-2.2.0/bin/htop
+</pre>
+
+<p>This is the path to the binary in the Nix store. Nix installs all packages into the store using a combination of its hash, name and version.</p>
+
+<p>You can search for available packages using <code>nix search</code>, for example:</p>
+
+<pre class="code">
+nix search python3
```

Comment:
`nix search` is experimental and is in the newest release hidden behind an option. Thus it cannot be "on the front page" yet.

Reply:
alright, will replace by `nix-env -qa …` then